src/robotFactory.js

```diff
@@ -1,10 +1,90 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    const { x = 0, y = 0 } = coords;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class DeliveryDrone {}
+    this.coords = {
+      x,
+      y,
+    };
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    const robotName = `Robot: ${this.name}`;
+    const chipVer = `Chip version: ${this.chipVersion}`;
+    const weight = `Weight: ${this.weight}`;
+
+    return `${robotName}, ${chipVer}, ${weight}`;
```

Comment:
```suggestion
    return `Robot: ${this.name}, Chip version: ${this.chipVersion},`
      +`Weight: ${this.weight}`;;
```
you can do something like that